tools/SourceKit/tools/sourcekitd/lib/API/CMakeLists.txt

```diff
@@ -7,11 +7,16 @@ set(sourcekitdAPI_sources
   TokenAnnotationsArray.cpp
 )
 
-set(sourcekitdAPI_Darwin_sources
-  sourcekitdAPI-XPC.cpp)
-set(LLVM_OPTIONAL_SOURCES ${sourcekitdAPI_Darwin_sources})
-if(APPLE)
-  list(APPEND sourcekitdAPI_sources ${sourcekitdAPI_Darwin_sources})
+set(sourcekitdAPI_InProc_sources sourcekitdAPI-InProc.cpp)
```

Comment:
I think it is important that the concept of "Darwin sources" be preserved here. When building on Mac, both the standard and in-proc versions of `sourcekitd` use XPC services to implement these APIs. In the case of in-proc library, `xpc_object_t` is effectively used as a flexible DTO to implement the SourceKit request object.

An alternative implementation will indeed be required though for Linux, perhaps using some sort of C++ variant type. I've just begun poking looking into what this could look like but haven't gotten very far.